packages/atomic/src/components/common/items-per-page/validate.ts

```diff
@@ -0,0 +1,32 @@
+import {AtomicCommerceProductsPerPage} from '../../commerce/atomic-commerce-products-per-page/atomic-commerce-products-per-page';
+import {AtomicResultsPerPage} from '../../search/atomic-results-per-page/atomic-results-per-page';
+
+export function validateChoicesDisplayed(
+  context: AtomicResultsPerPage | AtomicCommerceProductsPerPage
+) {
+  return context.choicesDisplayed.split(',').map((choice) => {
+    const parsedChoice = parseInt(choice);
+    if (isNaN(parsedChoice)) {
+      const errorMsg = `The choice value "${choice}" from the "choicesDisplayed" option is not a number.`;
+      context.bindings.engine.logger.error(errorMsg, context);
+      throw new Error(errorMsg);
+    }
+
+    return parsedChoice;
+  });
+}
```

Comment:
How about having this function takes choicesDisplayed, instead of the whole context?
Make it throw a dedicated Error (e.g. ChoiceIsNaNError), catch it, log it, rethrow it.

Reply:
This makes the initialize() function in the corresponding atomic components very verbose. I prefer it the way it is now. What exactly is the problem with passing the context ?

Reply:
agree !